client/src/lib/api/files.ts

```diff
@@ -32,6 +43,36 @@ export async function getSignedUploadURLs(user: User, fileNames: String[]) {
   return new Map<string, string>(Object.entries(result));
 }
 
+/**
+ * Get all files from the given dataset for the given user
+ *
+ * @param user User whose files need to be retrieved
+ * @returns A Promise that resolves to a list of UserFile objects
+ */
+export async function getFilesFromDataset(
+  user: User,
+  datasetName: string
+): Promise<UserFile[]> {
+  const getDataset = async () => {
```

Comment:
I would put this as a separate function in `lib/api/datasets.ts` that just gets called here.